h2/src/main/org/h2/table/MetaTable.java

```diff
@@ -1149,28 +1149,44 @@ private boolean hideTable(Table table, Session session) {
                 if (store != null) {
                     MVStore mvStore = store.getMvStore();
                     FileStore fs = mvStore.getFileStore();
-                    add(rows, "info.FILE_WRITE",
-                            Long.toString(fs.getWriteCount()));
-                    add(rows, "info.FILE_READ",
-                            Long.toString(fs.getReadCount()));
-                    add(rows, "info.UPDATE_FAILURE_PERCENT",
-                            String.format(Locale.ENGLISH, "%.2f%%", 100 * mvStore.getUpdateFailureRatio()));
-                    long size;
-                    try {
-                        size = fs.getFile().size();
-                    } catch (IOException e) {
-                        throw DbException.convertIOException(e, "Can not get size");
+                    if (fs != null) {
+                        add(rows, "info.FILE_WRITE",
+                                Long.toString(fs.getWriteCount()));
+                        add(rows, "info.FILE_WRITE_BYTES",
+                                Long.toString(fs.getWriteBytes()));
+                        add(rows, "info.FILE_READ",
+                                Long.toString(fs.getReadCount()));
+                        add(rows, "info.FILE_READ_BYTES",
+                                Long.toString(fs.getReadBytes()));
+                        add(rows, "info.UPDATE_FAILURE_PERCENT",
+                                String.format(Locale.ENGLISH, "%.2f%%", 100 * mvStore.getUpdateFailureRatio()));
+                        add(rows, "info.FILL_RATE",
+                                Integer.toString(mvStore.getFillRate()));
+                        add(rows, "info.CHUNKS_FILL_RATE",
+                                Integer.toString(mvStore.getChunksFillRate()));
+                        long size;
+                        try {
+                            size = fs.getFile().size();
+                        } catch (IOException e) {
+                            throw DbException.convertIOException(e, "Can not get size");
```

Comment:
I think the exception should be suppressed for safety.

Not really significant, but this code looks suspicious and raises attention.

Reply:
If file size can not be determined, we probably have a bigger problem at hand.

Reply:
I know, but if database isn't closed yet the statistics should be available for analysis.